core/components/molecules/tabs/Tabs.tsx

```diff
@@ -109,15 +109,132 @@ export const Tabs = (props: TabsProps) => {
+      setDisplayTabs(newList);
+    }
+  }, [showMore]);
+
+  React.useLayoutEffect(() => {
+    if (tabsWrapperRef.current) {
+      const containerClientWidth = tabsWrapperRef.current.clientWidth;
+      const containerScrollWidth = tabsWrapperRef.current.scrollWidth;
+
+      const isOverflow = containerScrollWidth > containerClientWidth;
+
+      setShowMore(isOverflow);
+      isOverflow && setWidthDifference(containerScrollWidth - containerClientWidth);
+    }
+  }, []);
+
+  const addMenuItemToTab = () => {
+    if (tabsWrapperRef.current) {
+      const containerClientWidth = tabsWrapperRef.current.clientWidth;
+      const containerScrollWidth = tabsWrapperRef.current.scrollWidth;
+
+      const isOverflow = containerScrollWidth > containerClientWidth;
+
+      setIsTabAdded(false);
+      isOverflow && setWidthDifference(containerScrollWidth - containerClientWidth);
+
+      let remainingWidth = containerScrollWidth - containerClientWidth;
+
+      if (isOverflow) {
+        let index = displayTabs.length - 1;
+        while (remainingWidth >= 0 && index > 0) {
+          remainingWidth = remainingWidth - allTabRefs[index - 1]?.scrollWidth;
+          index = index - 1;
+        }
+
+        const removeIndex = displayTabs.length - index;
+        const newList = displayTabs.slice(0, removeIndex).concat(displayTabs.slice(-1));
+
+        const extraList = [...tabs]
+          .slice(index, tabs.length)
+          .filter((item) => item !== displayTabs[displayTabs.length - 1]);
+
+        setExtraTabsList(extraList);
+        setDisplayTabs(newList);
+        setActiveTab(newList.length - 1);
+      }
+    }
+  };
+
+  const replaceTabWithMenuItem = () => {
+    if (tabsWrapperRef.current) {
+      const containerClientWidth = tabsWrapperRef.current.clientWidth;
+      const containerScrollWidth = tabsWrapperRef.current.scrollWidth;
+
+      const isOverflow = containerScrollWidth > containerClientWidth;
+
+      setShowMore(isOverflow);
+      setIsTabDismiss(false);
+      isOverflow && setWidthDifference(containerScrollWidth - containerClientWidth);
+
+      let remainingWidth = containerScrollWidth - containerClientWidth;
+
+      if (moreButtonRef.current?.clientWidth) {
+        remainingWidth = remainingWidth - moreButtonRef.current.clientWidth;
+      }
+
+      let index = tabs.length;
+      while (remainingWidth >= 0 && index >= 0) {
+        remainingWidth = remainingWidth - allTabRefs[index]?.scrollWidth;
+        index = index - 1;
+      }
+
+      const newList = [...tabs].slice(0, index + 1);
+      const extraList = [...tabs].slice(index + 1, tabs.length);
+      setExtraTabsList(extraList);
+      setDisplayTabs(newList);
+    }
+  };
+
+  React.useEffect(() => {
+    isTabDismiss && replaceTabWithMenuItem();
+    if (isTabAdded) {
+      const tabIndex = displayTabs.length - 1;
+      addMenuItemToTab();
+      setActiveTab(tabIndex);
+      if (onTabChange) onTabChange(tabIndex);
+    }
+  }, [displayTabs]);
```

Comment:
here, dependencies may not be clearly defined: please re-check
`isTabDismiss` and `isTabAdded` can be added here.